backend/db.js

```diff
@@ -3,46 +3,190 @@
-let client = null;
-
-// create a connection to url and call callback()
-function connect(url, callback) {
-    if (client == null) {
-        // create a mongodb client
-        client = new MongoClient(url, options);
-        // establish a new connection
-        client.connect((err) => {
-            if (err) {
-                // error occurred during connection
-                client = null;
-                callback(err);
-            } else {
-                // all done
-                callback();
-            }
-        });
-    } else {
-        // connection was established earlier. just call callback()
-        callback();
+
+class Database {
+    client = null;
+
+    // create a connection to url and call callback()
+    static connect(url, callback) {
+        if (Database.client == null) {
+            // create a mongodb client
+            Database.client = new MongoClient(url, options);
+            // establish a new connection
+            Database.client.connect((err) => {
+                if (err) {
+                    // error occurred during connection
+                    Database.client = null;
+                    callback(err);
+                } else {
+                    // all done
+                    callback();
+                }
+            });
+        } else {
+            // connection was established earlier. just call callback()
+            callback();
+        }
     }
-}
 
-// get database using pre-established connection
-function db(dbName) {
-    return client.db(dbName);
-}
+    static db(dbName) {
+        return Database.client.db(dbName);
+    }
 
-// close open connection
-function close() {
-    if (client) {
-        client.close();
-        client = null;
+    static close() {
+        if (Database.client) {
+            Database.client.close();
+            Database.client = null;
+        }
     }
-}
 
+    static store_user(body, next) {
+        return Database.db('GrandValet').collection('Users').findOne({username: body.username})
+        .then((user) => {
+            if (user == null) 
+            {
+                let newData = {type: body.type, username: body.username, name: body.name, password: body.password, email: body.email, phone: body.phone, driverStatus: body.driverStatus};
+                Database.db('GrandValet').collection('Users').insertOne(newData);
+                return;
+            }
+            Database.db('GrandValet').collection('Users').updateOne({username: body.username}, { $set: {type: body.type, name: body.name, password: body.password, email: body.email, phone: body.phone, driverStatus: body.driverStatus}});
+        });
+    }
+
+    static read_user(username, next) {
+        return Database.db('GrandValet').collection('Users').findOne({username: username})
+        .then((user) => {
+            if (user == null) 
+            {
+                console.log("bad")
+                return;

```

Comment:
Does this have the same effect as `return null`?

Reply:
Changed to `return null`